kt/godot-library/src/main/kotlin/godot/core/AABB.kt

```diff
@@ -396,6 +416,49 @@ class AABB(
         return under && over
     }
 
+    /**
+     * Returns `true` if the given ray intersects with this [AABB]. Ray length is infinite.
+     */
+    fun intersectsRay(from: Vector3, dir: Vector3): Boolean {
+        require(size.x >= 0 && size.y >= 0 && size.z >= 0) {
+            "AABB size is negative, this is not supported. Use AABB.abs() to get an AABB with a positive size."
+        }
```

Comment:
Is it a good idea to crash here? Shouldn't we log an error and return false instead?
I know godot "crashes" in such cases but don't we then have issues if such a call originated from the cpp side?

Same for other usages of `require` and the likes.

Reply:
I think we should better explain to jetbrains they have a vision regarding preprocessors that is not compatible with game industry.  
Here I would expect a debug version of engine to have this require call and crash, to spot problems asap, and a release version without this call.  
Anyway, this should not be modified in this PR IMO, as we have a lot of already existing require in core types.